Migrated PG long-running queries article, updated content, and added to toc file (#1367)
aiven/devportal: docs/products/postgresql/reference/pg-long-running-queries.rst

PostgreSQL® long-running queries
===============================

> Aiven does not terminate any customer queries even if they run indefinitely.  Long-running queries often cause issues by locking resources and preventing database maintenance tasks.

Reviewer: ```suggestion
Aiven does not terminate any customer queries even if they run indefinitely, but long-running queries can cause issues by locking resources and therefore preventing database maintenance tasks.
```

Author: added changes

To identify and terminate such long-running queries, you can do it from either:

> * `Aiven console <http://console.aiven.io>`_ / web page

Reviewer: ```suggestion
* `Aiven Console <http://console.aiven.io>`_
```

Author: added changes

* `PostgreSQL® shell <https://www.postgresql.org/docs/current/app-psql.html>`_ (``psql``)

> Aiven console

Reviewer: ```suggestion
Terminate long running queries from the Aiven Console
```

Author: added changes

^^^^^^^^^^^^^

> In the Aiven web console, you can go to the **Current Queries** tab for your service.

Reviewer: We usually refer to our console as the `Aiven Console` (check capitals) and always add the link to http://console.aiven.io

Author: added changes

.. image:: /images/products/postgresql/pg-long-running-queries.png
    :alt: PostgreSQL® service overview tab in Aiven's console

> On this page, you can click **Terminate** to stop any queries directly.

Reviewer: ```suggestion
On the Current Queries page, you can check the query duration and click **Terminate** to stop any long-running queries.
```

Author: added changes

> PostgreSQL® shell (psql)

Reviewer: ```suggestion
Detect and terminate long running queries with ``psql``
```

Author: added changes

^^^^^^^^^^^^^^^^^^^^^^^^

> You can login to your service by running on the terminal ``psql <service_uri>``.  Once connected, you can call the following function on the ``psql`` shell::

Reviewer: We can reference this article for the login part https://docs.aiven.io/docs/products/postgresql/howto/connect-psql.html

Author: added on the `login to your service` set of words

> You can login to your service by running on the terminal ``psql <service_uri>``.  Once connected, you can call the following function on the ``psql`` shell::

Reviewer: Also what is the `pg_terminate_backend(pid);` function doing, we should give the users a bit of explanation on what they're going to do before telling them to copy paste a function

Author: Essentially the [`pg_terminate_backend(pid)`](https://pgpedia.info/p/pg_terminate_backend.html) from my understanding, terminates a PG process/query.  I can link the official documentation for additional context and fixed the wording to match the original help article.

    SELECT pg_terminate_backend(pid);

You can use the following query to monitor currently running queries::

    SELECT * FROM pg_stat_activity WHERE state <> 'idle';

> Client applications can use the ``statement_timeout`` session variable to voluntarily request the server to automatically cancel any query using the current connection that runs over a specified length of time. For example, the following would cancel any query that runs for more 15 seconds automatically::

Reviewer: Can we put this as a Tip?

Author: This could be, but it would be a long Tip section since it would need the next 3 lines wouldn't it?

Author: @ftisiot do you still want this to be as a `Tip` section? It would a long one and it might not be able to cover the remaining content with different formats

Reviewer: let me have a look

    SET statement_timeout = 15000

You may check the `client connection defaults <https://www.postgresql.org/docs/current/runtime-config-client.html>`_ documentation for more information on the available session variables.

> Common errors

Reviewer: I believe we're only flagging a specific error, can we be a bit more precise with the title?

Author: You're right! Fixed that with `Common error` title and added the error message accordingly

^^^^^^^^^^^^^

> There may be a scenario where you encounter an error when running the above commands.

Reviewer: What is the error seen? is there a code people can reference?

Author: This is my bad, there's really only one error scenario provided and I failed to include that on the content.  Fixed that with the recent commit.

> If the database user is not a member of the database connected to, you may encounter the error::

Reviewer: This is a query, not an error?

Author: You're correct, and I failed to put the proper error message.  Fixed that on the recent commit.
